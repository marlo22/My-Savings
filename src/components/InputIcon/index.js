import React from 'react';
import PropTypes from 'prop-types';

import { Icon } from 'native-base';

const InputIcon = ({ status }) => {
  if (status) {
    return <Icon name="checkmark-circle" />;
  }

  if (status === false) {
    return <Icon name="close-circle" />
  }

  return null;
};

InputIcon.propTypes = {
  status: PropTypes.bool
};

export default InputIcon;