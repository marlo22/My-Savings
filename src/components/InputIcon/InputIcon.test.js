import React from 'react';
import { shallow } from 'enzyme';
import InputIcon from './';

const component = shallow(
  <InputIcon status={true} />
);

describe('<InputIcon />', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('if status is "true" should return checkmark-circle Icon', () => {
    expect(component.prop('name')).toBe('checkmark-circle');
  });

  test('if status is "false" should return close-circle Icon', () => {
    component.setProps({ status: false });

    expect(component.prop('name')).toBe('close-circle');
  });

  test('if status is "undefined" should return null', () => {
    component.setProps({ status: undefined });

    expect(component.html()).toBe(null);
  });
});