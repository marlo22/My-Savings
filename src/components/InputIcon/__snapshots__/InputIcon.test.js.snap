// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputIcon /> snapshot 1`] = `
<Styled(Icon)
  name="checkmark-circle"
/>
`;
